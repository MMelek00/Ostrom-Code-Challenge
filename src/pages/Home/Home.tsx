import { FC, memo } from "react";
import { RouteComponentProps } from "react-router-dom";
import { useSelector, shallowEqual } from "react-redux";
import { Helmet } from "react-helmet";

import { AppState } from "../../store";
import { StudentList } from "../../components";
import styles from "./styles.module.scss";

export type Props = RouteComponentProps;

const Home: FC<Props> = (): JSX.Element => {
  const studentsList = useSelector(
    (state: AppState) => state.studentList,
    shallowEqual
  );
  const renderList = () => <StudentList items={studentsList} />;

  return (
    <div className={styles.Home}>
      <Helmet title="Home" />
      {renderList()}
    </div>
  );
};
export default memo(Home);
