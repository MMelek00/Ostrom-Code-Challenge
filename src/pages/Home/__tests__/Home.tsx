import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import mockStore from "../../../utils/mockStore";
import Home from "../Home";

describe("<Home />", () => {
  const renderHelper = (reducer = [{
    "id": "Id1637586198021",
    "firstName": "melek",
    "lastName": "mliki",
    "birthday": "26/02/1996",
    "course": "ReactJs",
    "hours": "23",
    "price": "25"
  }]) => {
    const { dispatch, ProviderWithStore } = mockStore({ studentList: reducer });
    const { container } = render(
      <ProviderWithStore>
        <MemoryRouter>
          {/* 
            @ts-expect-error */}
          <Home />
        </MemoryRouter>
      </ProviderWithStore>
    );

    return { dispatch, firstChild: container.firstChild };
  };

  it("renders the <List /> if loading was successful", () => {
    const reducer = 
      [
        {
          "id": "Id1637586198021",
          "firstName": "melek",
          "lastName": "mliki",
          "birthday": "26/02/1996",
          "course": "ReactJs",
          "hours": "23",
          "price": "25"
        },
      ];

    expect(renderHelper(reducer).firstChild).toMatchSnapshot();
  });
});
