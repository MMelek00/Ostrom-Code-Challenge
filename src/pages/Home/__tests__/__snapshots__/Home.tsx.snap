// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders the <List /> if loading was successful 1`] = `
<div
  class="Home"
>
  <div
    class="StudentsList card"
  >
    <div
      class="card-body"
    >
      <button
        class="btn btn-success btn-sm"
        type="button"
      >
        Add Student
      </button>
      <table>
        <tbody>
          <tr>
            <th>
              First Name
            </th>
            <th>
              Last Name
            </th>
            <th>
              Date Of Birth
            </th>
            <th>
              Course
            </th>
            <th>
              Hours
            </th>
            <th>
              Price
            </th>
            <th />
            <th />
          </tr>
          <tr>
            <td>
              melek
            </td>
            <td>
              mliki
            </td>
            <td>
              26/02/1996
            </td>
            <td>
              ReactJs
            </td>
            <td>
              23
            </td>
            <td>
              25
            </td>
            <td>
               
              <button
                class="btn btn-outline-success btn-sm d-block w-100"
                type="button"
              >
                Edit
              </button>
            </td>
            <td>
               
              <button
                class="btn btn-outline-success btn-sm d-block w-100"
                type="button"
              >
                Delete
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
